mathesar_ui/src/systems/table-view/table-inspector/column/SetDefaultValue.svelte

```diff
@@ -9,21 +9,42 @@
   import { toast } from '@mathesar/stores/toast';
 
   export let column: ProcessedColumn;
+  export let canExecuteDDL: boolean;
 
   const tabularData = getTabularDataStoreFromContext();
   $: ({ columnsDataStore, recordsData } = $tabularData);
   $: ({ recordSummaries } = recordsData);
 
-  $: value = column.column.default?.value;
-  $: actionButtonsVisible = value !== column.column.default?.value;
+  $: initialValue = column.column.default?.value;
+  $: value = initialValue;
+
+  /**
+   * `value` is being passed to different types
+   * of input components by the `DynamicInput` component.
+   * All components when passed `value={undefined}` let it be undefined
+   * but input and textarea change it to empty string
+   * which makes the check `value===initialValue` to fail
+   * hence the sanitization.
+   */
+  $: sanitizedValue = value === '' ? undefined : value;
+
+  /**
+   * Not using strict equality here since
+   * numbers are being sent as string to the backend
+   * but the initialValue from BE is always of the
+   * correct data type.
+   */
+  // eslint-disable-next-line eqeqeq
+  $: actionButtonsVisible = sanitizedValue != initialValue;
```

Comment:
There is a `initialInputValue` property in ProcessedColumn which should provide the value of the input when the default value is undefined. This eliminates the need for sanitizedValue.

I'm recommending a value & initialValue comparison with stringified values via JSON.stringify since this should provide the value as returned from the server. This is more versatile and works with types like JSON map, JSON list, and array types. The current check would not work for any of them.

```suggestion
  $: initialValue = column.column.default?.value ?? column.initialInputValue;
  $: value = initialValue;
  $: actionButtonsVisible = JSON.stringify(value) !== JSON.stringify(initialValue);
```

Reply:
@pavish 

>There is an initialInputValue property in ProcessedColumn which should provide the value of the input when the default value is undefined. This eliminates the need for sanitized value.

Could not find this key inside the ProcessedColumn instead using: `getInitialInputValue(column.column);`

>I'm recommending a value & initialValue comparison with stringified values via JSON.stringify since this should provide the value as returned from the server. This is more versatile and works with types like JSON map, JSON list, and array types. The current check would not work for any of them.

There are two reasons for me not taking this approach

**First Reason:** 
When setting a default value for a column with data type number, let's say `value` will be `"1145"` and `initialValue` will be `1145`.
```typescript

$: value = "1145";
$: initialValue = 1145;
$: actionButtionsVisible = JSON.stringify(value) !== JSON.stringify(initialValue);

```

Here actionButtionsVisible will be false should be true since `JSON.stringify("1145") = "\"1145\""` and `JSON.stringify(1145) = "1145"`.

**Second Reason:**
We do not have any such column types as of now that will need JSON map, JSON list, or array types and the purpose of this PR is very different

Reply:
@rajatvijay 

> Could not find this key inside the ProcessedColumn instead using: getInitialInputValue(column.column);

It's worth adding it to ProcessedColumn. You can refer data explorer which has it.

> We do not have any such column types as of now that will need JSON map, JSON list, or array types and the purpose of this PR is very different

We do have all of these column types implemented, please refer the cell-fabric component. I have disabled it while creating a column for the release. The user is still able to convert a string column to JSON list or map, and the database the user connects to can have these columns.

> Here actionButtionsVisible will be false should be true since JSON.stringify("1145") = "\"1145\"" and JSON.stringify(1145) = "1145".

I'd suggest a type check before this check, instead of `==`. For literals, we could do a `String(value) === String(initialValue)` and we can do a `JSON.stringify` check for objects.

Reply:
385b210dd

Reply:
Looks good